File: Second ATM project.cpp
#include <iostream>
using namespace std;


//************ Variables ***********//
float balance = 1000;
int deposit = 0;
int withdraw = 0;
string password = "0"; //the password is string because the errors of entering a string in a int variable
string input_password; 
int choice; //choice of menu
int attempts = 0; //attempts of password 
int language_choice;
//*********** End Variables **********//


void show (){
  //main menu
cout<<"\n************** Menu **************"<<endl;
cout<<"1:Balance"<<endl;
cout<<"2:Withdraw"<<endl;
cout<<"3:Deposit"<<endl; 
cout<<"4:Exit"<<endl;
cout<<"**********************************"<<endl;
}


bool language() {
  //language menu
    cout << "********* select language *********" << endl;
    cout<<"1:English"<<endl;
    cout<<"2:Arabic"<<endl;
    cout<<"3:Exit"<<endl;
    cout << "***********************************" << endl;

   //language choice input 
    while (true) {
        cin >> language_choice;
        cout<<endl;
        //checking if it's int
        if (cin.fail()) { //this if (cin.fail()) from chatGPT ,
        //to handle errors if the input was string
            cin.clear();
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
            cout << "Please enter a valid number from the (select language) menu : ";
        }
        
        
        // English language 
        else if (language_choice == 1) {
            return true;
        }
        
        
        // Arabic language 
        else if (language_choice == 2) {
          cout<<"Arabic language is not available at the moment please try again later, or chose another language from (select language) menu : ";
        }
        
        
        //Exit choice 
        else if (language_choice == 3){
          cout<<"Thank you"<<endl;
          return false ; 
        }
        
        
        //if language choice input wasn't int
        else {
            cout << "Please enter a vaild number from the (select language) menu : ";
        }
    } // End while (true)
}


// password function
bool pin (){
    
    cout<<"please Enter your password : ";
    cin>>input_password;
    cout<<endl;
    do{ 
    if (password == input_password)  
    {
        return true;
    }
    else
    {
        cout<<"your password is incorrect, please try again : ";
        cin>>input_password;
        cout<<endl;
        if(password==input_password)
        {
            return true;
        }
        ++attempts;                      
    }
    }while (attempts < 2);
    
    cout<<"your card will be temporarily blocked for 24 hours"<<endl;
    return false;
}


//process function where the menu choice is
void process (){
  do{
    cout<<"Enter your choice : ";
    cin>>choice;
    if (cin.fail()) {
    cin.clear();
    cin.ignore(numeric_limits <streamsize> :: max(), '\n');
  }
    cout<<endl;
    switch(choice){
      
      case 1:
      cout<<"Your balance is "<<balance<<endl;
      cout<<endl;
      break;
      
      
      case 2:
      cout<<"Enter the amount : ";
      cin>>withdraw;
      cout<<endl;
      if (cin.fail()) {
        cin.clear();
        cin.ignore(numeric_limits <streamsize> :: max(), '\n');
        cout<<"Please enter a vaild number from the menu"<<endl;
        cout<<endl;
        }
        else {
      if (balance>=withdraw){
      balance -= withdraw;
      cout<<"Now : your balance is "<<balance<<endl;
      cout<<endl;
      }
      else
      {
        cout<<"Sorry your balance is less than your withdraw"<<endl;
        cout<<endl;
      }
        }
  break;
      
      
      case 3:
      cout<<"put the money that you need to deposit : ";
      cin>>deposit;
      cout<<endl;
      if (cin.fail()) {
        cin.clear();
        cin.ignore(numeric_limits <streamsize> :: max(), '\n');
        cout<<"Please enter a vaild number from the menu"<<endl;
        cout<<endl;
        }
        else {
      balance += deposit;
      cout<<"Now : your balance is "<<balance<<endl;
      cout<<endl;
        }
      break;
      
      
      case 4:
      cout<<"Thank you"<<endl;
      return;
      break;
      
      
      default :
      cout<<"Please enter a vaild number from the menu"<<endl;
      cout<<endl;
      
      
    } //End switch 
  }while (true);
  }

  
int main (){
  
  if (language () == false){
    return 0;
  }
  if (pin() == false ){
    return 0;
  }             
  show ();
  process ();
//I don't know if the PIN should be first :)

  return 0;
}

File: Second ATM project.py

#//$$$$$$$$$$$$ THIS CODE WAS WRITTEN ON ANDROID DEVICE $$$$$$$$$$$$\\#

#it's my first project in python 
# I used "Acode" for writing this code
#It's not that hard coding on android but still weird :|


#//************ Variables ***********//
balance = 1000
deposit = 0
withdraw = 0
password = "0"
input_password = None
choice = None
attempts = 0
language_choice = None
#//*********** End Variables **********//


def show () :
  print ("""\n************** Menu **************
1:Balance
2:Withdraw
3:Deposit
4:Exit
***********************************""")


def language () :
  print("""********* select language *********
1:English
2:Arabic
3:Exit
************************************""")

  while True : 
    language_choice = input()
    if language_choice == "1" :
      return True
    
    elif language_choice == "2" :
      print ("Arabic language is not available at the moment please try again later, or chose another language from (select language) menu : ")
    
    elif language_choice == "3" :
      print ("Thank you")
      return False
    
    else : 
      print ("Please enter a vaild number from the (select language) menu : ")
    
    
def pin () :
  global attempts 
  input_password = input("please Enter your password : ")
  if input_password == password :
    return True
    
  else :
    while input_password != password and attempts < 2 :
      
      input_password = input("your password is incorrect, please try again : ")
      attempts = attempts + 1
      if input_password == password :
        return True
        
  print ("your card will be temporarily blocked for 24 hours")
  return False


def process () :
  global balance
  while True :
    
    choice = input("Enter your choice : ")
    if choice == "1" :
      print ("you balance is ",balance)
      
    elif choice == "2" :
      
      withdraw = int(input("Enter the amount : "))
      if balance >= withdraw :
        balance = balance - withdraw
        print("Now : your balance is ",balance)
      else :
        print ("Sorry your balance is less than your withdraw")
        
    elif choice == "3" :
        
        deposit = int(input ("put the money that you need to deposit : "))
        balance = balance + deposit
        print ("Your balance is ",balance)
      
    elif choice == "4" :
        print ("Thank you")
        return False
    
    else :
        print ("Please enter a vaild number from the menu")


if language () == True :
  if pin () == True :
    show ()
    process ()
  